sdks/java/core/src/main/java/org/apache/beam/sdk/util/PubsubJsonClient.java

```diff
@@ -135,7 +135,7 @@ public int publish(TopicPath topic, List<OutgoingMessage> outgoingMessages)
     for (OutgoingMessage outgoingMessage : outgoingMessages) {
       PubsubMessage pubsubMessage = new PubsubMessage().encodeData(outgoingMessage.elementBytes);
 
-      Map<String, String> attributes = pubsubMessage.getAttributes();
+      Map<String, String> attributes = outgoingMessage.attributes;
```

Comment:
formerly, the `PubSub` message attributes would always be null, yes?

Is there any defined behavior about `timestampLabel` and `idLabel` not already being in the attributes? We'll overwrite them here if they're already specified.